File: 2021/d01.py
with open('i01.txt') as file:
    data = list(map(int, file.readlines()))

print('Part 1:', [data[i-1] < data[i]
      for i in range(1, len(data))].count(True))
print('Part 2:', [data[i-3] < data[i]
      for i in range(3, len(data))].count(True))

File: 2021/d01.dart
import 'dart:io';

void main() {
  // Part 1
  print(File('i01.txt').readAsLinesSync().map(int.parse).fold<List<num>>(
      [0, double.infinity], (pr, e) => [pr[0] += (pr[1] < e ? 1 : 0), e])[0]);
}
